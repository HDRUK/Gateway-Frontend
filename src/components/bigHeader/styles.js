import styled from "styled-components";
import { HeaderWrapper } from "../../styles/carbonComponents.js";

export const BigHeaderWrapper = styled(HeaderWrapper)`
    display: block;
    height: 33.5rem;
    padding-top: 2rem;
`;

export const BigHeaderImage = styled.img`
    height: 6.4375rem;
    display: block;
`;
