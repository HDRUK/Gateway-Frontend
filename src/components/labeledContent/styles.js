import styled from "styled-components";

export const LabeledContentWrapper = styled.div`
    display: inline-block;

    h2,
    p {
        padding-top: 0.125rem;
        padding-right: 0.5rem;
    }
`;

export const HeaderLabel = styled.h2`
    font-size: 0.625rem;
    line-height: 0.75rem;
    color: #979797;
    text-transform: uppercase;
`;

export const ContentText = styled.p`
    font-size: 0.875rem;
    line-height: 1.0625rem;
    color: #0f62fe;

    ${p => !p.lowercase && "text-transform: uppercase"}
`;
