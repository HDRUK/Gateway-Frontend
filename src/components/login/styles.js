import styled from "styled-components";

export const LoginWrapper = styled.span`
    padding: 0 2rem;
    text-align: center;
`;
