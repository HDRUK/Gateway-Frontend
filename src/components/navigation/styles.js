import styled from "styled-components";
import { HeaderNavigation } from "carbon-components-react";

export const NavigationWrapper = styled.div`
    width: 100%;
    height: ${p => p.theme.sizes.navigation}rem;
    border-bottom: 0.0625rem solid ${p => p.theme.colors.border};
`;

export const NavigationHeader = styled(HeaderNavigation)`
    :before {
        content: unset;
    }

    @media (max-width: 66rem) {
        display: block;
    }
`;

export const NavigationItem = styled.div`
    height: 100%;
    min-height: 2.875rem;
    padding: 1rem;
    line-height: 0.875rem;
    text-decoration: none;
    color: ${p => p.theme.text.primary};

    ${p => p.active && `border-bottom: 0.25rem solid ${p.theme.colors.primarySolid}`};
    font-weight: ${p => p.active && p.theme.text.bold};

    :hover,
    :active {
        border-bottom: 0.25rem solid ${p => p.theme.colors.primarySolid};
        font-weight: ${p => p.theme.text.bold};
    }
`;
