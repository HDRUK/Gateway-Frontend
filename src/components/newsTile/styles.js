import styled from "styled-components";

export const NewsTileWrapper = styled.div`
    display: inline-block;
    width: 26rem;
    height: 27.5rem;
    overflow: hidden;
    border-radius: 0.125rem;
    box-shadow: 0 0 0.25rem 0 ${p => p.theme.colors.shadow};
`;
