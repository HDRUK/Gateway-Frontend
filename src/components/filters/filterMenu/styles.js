import styled from "styled-components";
import { AccordionBlock } from "../../../styles/carbonComponents";

export const ExpandingAccordionBlock = styled(AccordionBlock).attrs(p => ({
    style: {
        paddingBottom: p.expandheight ? `${p.expandheight}px` : "1rem"
    }
}))`
    display: inline-block;
    padding-bottom: "1rem";
`;
