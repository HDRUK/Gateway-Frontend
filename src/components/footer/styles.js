import styled from "styled-components";
import { WidthWrapper, Logo } from "../../styles/styles.js";

export const FooterWrapper = styled.div`
    height: ${p => p.theme.sizes.footer}rem;
    background: linear-gradient(135deg, ${p => p.theme.colors.primary}, ${p => p.theme.colors.secondary});
    color: white;
    padding-top: 4rem;
    padding-bottom: 1rem;
    font-size: 1.1rem;
    position: relative;
`;

export const FooterContent = styled(WidthWrapper)`
    display: block;
    margin: 0 auto 0;
    max-width: 62rem;
    padding: 0 2rem;
`;

export const FooterBlock = styled.div`
    display: inline-block;
    box-sizing: border-box;
    width: 33.33%;
    height: 100%;
    vertical-align: top;
    padding: 0 4rem 0 0;
    line-height: 1.1875rem;

    :last-of-type {
        padding-right: 0;
        padding-left: 4rem;
    }
`;

export const FooterText = styled.div`
    width: 100%;
    margin: 0 0 0.4rem 0;
`;

export const FooterImage = styled(Logo)`
    max-width: 8rem;
    max-height: 5rem;
`;
