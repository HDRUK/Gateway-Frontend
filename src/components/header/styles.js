import styled from "styled-components";
import { LinkNoDecoration } from "../../styles/styles.js";

export const HeaderAlignment = styled.div`
    width: 100%;
`;

export const HeaderAlignmentItem = styled.div`
    height: 100%;
    float: ${p => (p.right ? "right" : "left")};
`;

export const HeaderSpacer = styled.div`
    height: 100%;
    width: 6rem;
`;

export const HeaderLink = styled.a`
    :hover {
        text-decoration: underline;
        text-decoration-color: ${p => p.theme.colors.white};
    }
`;
