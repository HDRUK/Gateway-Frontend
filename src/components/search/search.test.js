import React from "react";
import { create } from "react-test-renderer";
import Search from "./search";
import { ParagraphHeading, CenterBlock, LargeSpace, SmallSpace } from "../../styles/styles.js";
import { SearchBar } from "../../styles/carbonComponents.js";

describe("<Search> ", () => {
    let renderedComponent;

    beforeEach(() => {
        renderedComponent = create(<Search />);
    });

    it("should test the NewsTile Snapshot", () => {
        expect(renderedComponent).toMatchSnapshot();
    });

    it("should check the correct style elements are displayed", () => {
        const paragraphText = renderedComponent.root.findAllByType(ParagraphHeading);
        const darkText = renderedComponent.root.findAllByType(CenterBlock);
        const largeSpace = renderedComponent.root.findAllByType(LargeSpace);
        const smallSpace = renderedComponent.root.findAllByType(SmallSpace);
        const searchBar = renderedComponent.root.findAllByType(SearchBar);

        expect(paragraphText).toHaveLength(1);
        expect(darkText).toHaveLength(1);
        expect(largeSpace).toHaveLength(1);
        expect(smallSpace).toHaveLength(1);
        expect(searchBar).toHaveLength(1);
    });
});
