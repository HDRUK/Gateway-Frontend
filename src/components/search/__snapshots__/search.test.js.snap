// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search>  should test the NewsTile Snapshot 1`] = `
<div
  className="sc-gzVnrw ddkUmy"
>
  <div>
    <div
      className="sc-iwsKbI eSJicE"
    />
    <h2
      className="sc-EHOje btWVAQ"
    >
      What health data do you need? 
    </h2>
    <div
      className="sc-htoDjs dPlVZm"
    />
    <div
      className="bx--search bx--search--xl sc-kEYyzF gfWKdx"
    >
      <svg
        aria-hidden={true}
        className="bx--search-magnifier"
        focusable="false"
        height={16}
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "willChange": "transform",
          }
        }
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 14.3L10.7 10c1.9-2.3 1.6-5.8-.7-7.7S4.2.7 2.3 3 .7 8.8 3 10.7c2 1.7 5 1.7 7 0l4.3 4.3.7-.7zM2 6.5C2 4 4 2 6.5 2S11 4 11 6.5 9 11 6.5 11 2 9 2 6.5z"
        />
      </svg>
      <label
        className="bx--label"
        htmlFor="search__input__id_gtc5bjv97x"
      >
        Search
      </label>
      <input
        className="bx--search-input"
        id="search__input__id_gtc5bjv97x"
        onChange={[Function]}
        placeholder=""
        type="text"
      />
      <button
        aria-label="Clear search input"
        className="bx--search-close bx--search-close--hidden"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          focusable="false"
          height={20}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "willChange": "transform",
            }
          }
          viewBox="0 0 32 32"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
